federation/pkg/dnsprovider/dns.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsprovider
+
+import "k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+
+// Interface an abstract, pluggable interface for DNS providers.
+type Interface interface {
```

Comment:
The qualified name used by clients is "dnsprovider.Interface" which seems better than dnsprovider.DNS?  I'm inclined to leave it the way it is.  "Interface" is also consistent with the cloudprovider interface, I think,